ytdl_nfo/Ytdl_nfo.py

```diff
@@ -14,14 +14,17 @@ def __init__(self, file_path, extractor=None):
         self.extractor = extractor
         if extractor is None:
             self.extractor = self.data['extractor'].lower()
-
-        self.filename = os.path.splitext(self.data['_filename'])[0]
+        
+        if file_path.endswith(".info.json"):
+            self.filename = file_path[:-10]
+        else:
+            self.filename = os.path.splitext(self.data['_filename'])[0]
```

Comment:
The `_filename` field in my info files cannot reliably be used this way. I'm guessing because videos and json files are arranged into uploader subfolders in my case (e.g. `videos/[uploader]/[title]`) but were not in yours?

I don't know if falling back to the old behavior entirely makes sense, but anyway it works for me with this change.